### PR TITLE
TUM-29 Database setup

### PR DESCRIPTION
**Pull request by guustek**
- Added repository implementations for all entities.
- Added EF Core models configuration.
- Added unit of work pattern to commands.
- Adjusted Dapper queries to work with PostgreSQL, identifiers had to be quoted since EF Core is configured to create table names matching entity class names.
- Remove in memory repositories test implementations
- Changed all collection properties to empty by default instead of nullable in entities

### CODE REVIEW

File: Backend/Tumultu/Application/AnalysisResults/Commands/CreateAnalysisResultCommand.cs
Comment (Camed) [inline Backend/Tumultu/Application/AnalysisResults/Commands/CreateAnalysisResultCommand.cs:1]: We are not using _Tumultu.Domain.Entities_ here (yet, as we are throwing exception), but as a architecture convention I guess we should keep it here as using.

File: Backend/Tumultu/Application/Behaviours/Commands/CreateBehaviourCommand.cs [hunk 1]
```diff
@@ -1,6 +1,5 @@
 ﻿using MediatR;
 using Tumultu.Application.Common.Interfaces;
-using Tumultu.Domain.Entities;
```
Comment (Camed) [inline Backend/Tumultu/Application/Behaviours/Commands/CreateBehaviourCommand.cs]: Same here as previously, I would leave it for now

File: Backend/Tumultu/Infrastructure/Database/DependencyInjection.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+using Tumultu.Infrastructure.Database.Dapper.Repositories.Files;
+using Tumultu.Infrastructure.Database.Dapper.Repositories.FileVariants;
+using Tumultu.Infrastructure.Database.Dapper.Repositories.Users;
+using Tumultu.Infrastructure.Database.EFCore;
+using Tumultu.Infrastructure.Database.EFCore.Repositories.AnalysisResults;
+using Tumultu.Infrastructure.Database.EFCore.Repositories.Behaviours;
+using Tumultu.Infrastructure.Database.EFCore.Repositories.Common;
+using Tumultu.Infrastructure.Database.EFCore.Repositories.Files;
+using Tumultu.Infrastructure.Database.EFCore.Repositories.FileVariants;
+using Tumultu.Infrastructure.Database.EFCore.Repositories.Users;
```
Comment (Camed) [inline Backend/Tumultu/Infrastructure/Database/DependencyInjection.cs:22]: XD nothing wrong, but I love that walltext of usings

**Review by Camed — APPROVED** [review]
👌

**Review by Camed — APPROVED** [review]
👌

File: Backend/Tumultu/Application/AnalysisResults/Commands/CreateAnalysisResultCommand.cs
Comment (guustek) [reply Backend/Tumultu/Application/AnalysisResults/Commands/CreateAnalysisResultCommand.cs:1]: Ok, will revert this removal

File: Backend/Tumultu/Application/Behaviours/Commands/CreateBehaviourCommand.cs (hunk 1 above)
Comment (guustek) [reply Backend/Tumultu/Application/Behaviours/Commands/CreateBehaviourCommand.cs]: Ok, will revert this removal

File: Backend/Tumultu/Infrastructure/Database/DependencyInjection.cs (hunk 2 above)
Comment (guustek) [reply Backend/Tumultu/Infrastructure/Database/DependencyInjection.cs:22]: Will adjust that to "using whole namespace" I guess

**Review by Camed — COMMENTED** [review]
👍👍👍🔥🔥🔥🧯🧯🧯

**Pull request merged**